examples/dreambooth_custom_character/README.md

```diff
@@ -0,0 +1,200 @@
+pipe = StableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4")
+pipe.to("cuda") # If using CUDA for GPU
+
+def generate_images(prompt):
+    images = pipe(prompt, guidance_scale=7.5, num_images_per_prompt=4).images
+    return images 
+
+with gr.Blocks() as demo: 
+	prompt = gr.components.Textbox(label="Enter Prompt")
+	generate = gr.Button("Generate candidate images") 
+	images = {}
+	with gr.Row()
+		for i in range(1,5):
+			with gr.Column():
+				images[i] = gr.components.Image(label=f"Candidate Image {i}", type='pil')
+	generate.click(generate_images, inputs=prompt, outputs=list(images.values()))
+
+demo.launch(share=True)
+
+```
+
+This yields us the following UI, which we can use to generate some sample results for the base model (we used CompVis/stable-diffusion-v1-4, but feel free to experiment with others) across 3 prompting strategies. We’ll revisit these three prompts after our fine-tuning step to see what the model has learned!
+
+![](images/happy-fluffy-before.png)
+![](images/oxenai-ox-before.png)
+![](images/plain-ox-before.png)
+
+### Fine-tuning with Dreambooth
+
+The model training script we'll use is in the `diffusers` git repository we cloned earlier, at path `diffusers/examples/dreambooth/train_dreambooth.py`.
+
+We’ll first set a few variables to correctly parameterize the script for our custom use case:
+
+- `MODEL_NAME`: base model to start with from Hugging Face
+- `INSTANCE_DIR`: directory containing our imagery for fine-tuning - will point to the `images` folder in our oxen repo
+- `OUTPUT_DIR`: name of the model output folder (and the model name if uploading to Huggingface)
+- `INSTANCE_PROMPT`: the unique “name” of the subject we want the model to learn
+
+```bash
+export MODEL_NAME = "CompVis/stable-diffusion-v1-4"
+export INSTANCE_DIR = "./dreambooth-ox/images"
+export OUTPUT_DIR = "stable-diffusion-oxified"
+export INSTANCE_PROMPT = "an image of the oxenai ox"
+```
+
+We’re now set to run the script. All below flags use Hugging Face’s recommended settings. You’ll be directed by the CLI to authenticate with Hugging Face.
+
+```bash
+accelerate launch diffusers/examples/dreambooth/train_dreambooth.py \
+  --pretrained_model_name_or_path=$MODEL_NAME  \
+  --instance_data_dir=$INSTANCE_DIR \
+  --output_dir=$OUTPUT_DIR \
+  --instance_prompt="$INSTANCE_PROMPT" \
+  --resolution=512 \
+  --train_batch_size=1 \
+  --gradient_accumulation_steps=1 \
+  --learning_rate=5e-6 \
+  --lr_scheduler="constant" \
+  --lr_warmup_steps=0 \
+  --max_train_steps=400 \
+  --push_to_hub
+```
+
+This took around 5 minutes to run on a single NVIDIA A10 with 24GB of GPU VRAM, which we rented from [Lambda Labs](https://lambdalabs.com/).
+
+### Generating new images 
+If the `--push_to_hub` flag was set, the script will have pushed the resulting model up to Hugging Face after training. As such, we can modify our inference UI code as following to run inference on the new model: 
```

Comment:
Eventually.....I think we should be blog storage for models as well. But this is fine for now.

Reply:
agreed